src/NuGet.Clients/NuGet.VisualStudio.OnlineEnvironment.Client/SolutionExplorer/PackageManagerUICommandHandler.cs

```diff
@@ -247,41 +250,69 @@ private async ValueTask<IVsWindowFrame> CreateToolWindowAsync(WorkspaceVisualNod
                 throw new InvalidOperationException();
             }
 
-            var projectContextInfo = await ProjectContextInfo.CreateAsync(projectGuid.ToString(), CancellationToken.None);
-            var uiController = UIFactory.Value.Create(projectContextInfo);
-            var model = new PackageManagerModel(uiController, isSolution: false, editorFactoryGuid: GuidList.NuGetEditorType);
-            var control = await PackageManagerControl.CreateAsync(model, OutputConsoleLogger.Value);
-            var caption = string.Format(CultureInfo.CurrentCulture, Resx.Label_NuGetWindowCaption, Path.GetFileNameWithoutExtension(workspaceVisualNodeBase.NodeMoniker));
+            IVsWindowFrame windowFrame = null;
 
             var uiShell = await _asyncServiceProvider.GetServiceAsync(typeof(SVsUIShell)) as IVsUIShell;
             Assumes.Present(uiShell);
 
-            int[] pfDefaultPosition = null;
-            IVsWindowFrame windowFrame;
-            var windowPane = new PackageManagerToolWindowPane(control);
-            ErrorHandler.ThrowOnFailure(
-                uiShell.CreateToolWindow(
-                    (uint)__VSCREATETOOLWIN.CTW_fInitNew,
-                    0,              // dwToolWindowId - singleInstance = 0
-                    windowPane,     // ToolWindowPane
-                    Guid.Empty,     // rclsidTool = GUID_NULL
-                    projectGuid,
-                    Guid.Empty,     // reserved - do not use - GUID_NULL
-                    null,           // IServiceProvider
-                    caption,
-                    pfDefaultPosition,
-                    out windowFrame));
-
-            if (windowFrame != null)
+            uint toolWindowId;
+            bool foundToolWindowId = _projectGuidToToolWindowId.TryGetValue(projectGuid.ToString(), out toolWindowId);
+            const uint FTW_none = 0;
+
+            if (foundToolWindowId)
             {
-                WindowFrameHelper.AddF1HelpKeyword(windowFrame, keywordValue: F1KeywordValuePmUI);
-                WindowFrameHelper.DisableWindowAutoReopen(windowFrame);
-                WindowFrameHelper.DockToolWindow(windowFrame);
+                uiShell.FindToolWindowEx(
```

Comment:
Why are you using Ex here?

Reply:
so we can pass in the toolWindowId and get back that frame. per Andrew's advice.